### DISCUSSION TITLE
[globset] Match a folder and its children using a single glob

### DISCUSSION THREAD

**Discussion by gyk** [BurntSushi/ripgrep#2118] [Q&A]
It is not uncommon that we want to filter a folder itself and all of its children. But after https://github.com/BurntSushi/ripgrep/pull/1756 was merged, we have to use two glob patterns to match the paths:

```
/path/to/folder
/path/to/folder/**
```

Are there any more elegant ways to achieve this?

**Comment by BurntSushi** [comment]
Maybe `/path/to/folder{,/**}`?

It would help if you gave a concrete example. e.g., A directory tree along with the command you're running, the desired result and the actual result.

**Comment by gyk** [comment]
An example: I would like to ignore every `.AppleDouble` folder, which doesn't necessarily appear at the root, and all the items inside it. Unfortunately, appending the path with `{,/**}` still cannot pass the unittest below, unless I specify the exact version `globset = "=0.4.6"`.

```rust
#[cfg(test)]
mod tests {
    use globset::*;

    #[test]
    fn suffix() -> Result<(), Error> {
        // "**/target/**" doesn't work either
        let glob = GlobBuilder::new("**/target{,/**}")
            .build()?
            .compile_matcher();

        assert!(glob.is_match("target")); // failed
        assert!(glob.is_match("/target")); // failed
        assert!(glob.is_match("target/"));
        assert!(glob.is_match("/path/to/target")); // failed
        assert!(glob.is_match("/path/to/target/"));
        assert!(glob.is_match("/path/to/target/folder"));
        assert!(glob.is_match("/path/to/target/folder/"));

        assert!(!glob.is_match("targetfolder"));

        Ok(())
    }
}
```
